components/eamxx/src/control/atmosphere_driver.cpp

```diff
@@ -1283,6 +1288,56 @@ void AtmosphereDriver::set_initial_conditions ()
     }
   }
 
+  // Compute IC perturbations of GLL fields (if requested)
+  using vos = std::vector<std::string>;
+  const auto perturbed_fields = ic_pl.get<vos>("perturbed_fields", {});
+  const auto num_perturb_fields = perturbed_fields.size();
+  if (num_perturb_fields > 0) {
+    m_atm_logger->info("    [EAMxx] Adding random perturbation to ICs ...");
+
+    EKAT_REQUIRE_MSG(m_field_mgrs.count("Physics GLL") > 0,
+                     "Error! Random perturbation can only be applied to fields on "
+                     "the GLL grid, but no Physics GLL FieldManager was defined.\n");
+    const auto& fm = m_field_mgrs.at("Physics GLL");
+
+    // Setup RNG. If no random seed is requested in
+    // initial_conditions params, generate one using rand().
+    const auto seed = ic_pl.get<int>("perturbation_random_seed", rand());
+    std::mt19937_64 engine(seed);
+    m_atm_logger->info("      For IC perturbation, random seed: "+std::to_string(seed));
+
+    // Get perterbation limit. Defines a range [-perturbation_limit, perturbation_limit]
+    // for which the perturbation value will be randomly generated from.
+    const auto perturbation_limit = ic_pl.get<Real>("perturbation_limits", 0.001);
+
+    // Define a level mask using reference pressure and the pressure_lower_bound parameter.
+    // This mask dictates which levels we apply a perturbation.
+    const auto gll_grid = m_grids_manager->get_grid("Physics GLL");
+    gll_grid->get_geometry_data("hyam").sync_to_host(), gll_grid->get_geometry_data("hybm").sync_to_host();
+    const auto hyam = gll_grid->get_geometry_data("hyam").get_view<const Real*, Host>();
+    const auto hybm = gll_grid->get_geometry_data("hybm").get_view<const Real*, Host>();
+    constexpr auto ps0 = physics::Constants<Real>::P0;
+    const auto pressure_lower_bound = ic_pl.get<Real>("perturbation_pressure_lower_bound", 1050.0);
```

Comment:
kind of a dumb comment, but I initially understood "pressure_lower_bound" as saying you're going to use all pressures above a certain geometric height level instead of all pressures larger (i.e. closer to the surface) than this pressure level. Maybe change this name to min_perturbed_pressure or something?